Buttplug.Server.Managers.WinUSBManager/RezTranceVibratorDevice.cs

```diff
@@ -0,0 +1,92 @@
+{
+    class RezTranceVibratorDevice : ButtplugDevice
+    {
+        private USBDevice _device;
+        private static uint _vibratorCount = 1;
+
+        public RezTranceVibratorDevice(IButtplugLogManager aLogManager, USBDevice aDevice, uint aIndex)
+            : base(aLogManager, "Trancevibrator " + aIndex, "Trancevibrator " + aIndex)
+        {
+            _device = aDevice;
+            MsgFuncs.Add(typeof(SingleMotorVibrateCmd), new ButtplugDeviceWrapper(HandleSingleMotorVibrateCmd));
+            MsgFuncs.Add(typeof(StopDeviceCmd), new ButtplugDeviceWrapper(HandleStopDeviceCmd));
+            MsgFuncs.Add(typeof(VibrateCmd), new ButtplugDeviceWrapper(HandleVibrateCmd));
+        }
+
+        private Task<ButtplugMessage> HandleStopDeviceCmd(ButtplugDeviceMessage aMsg)
+        {
+            BpLogger.Debug("Stopping Device " + Name);
+            return HandleSingleMotorVibrateCmd(new SingleMotorVibrateCmd(aMsg.DeviceIndex, 0, aMsg.Id));
+        }
+
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+        private async Task<ButtplugMessage> HandleVibrateCmd(ButtplugDeviceMessage aMsg)
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
+        {
+            if (!(aMsg is VibrateCmd cmdMsg))
+            {
+                return BpLogger.LogErrorMsg(aMsg.Id, Error.ErrorClass.ERROR_DEVICE, "Wrong Handler");
+            }
+
+            if (cmdMsg.Speeds.Count == 0 || cmdMsg.Speeds.Count > _vibratorCount)
+            {
+                return new Error(
+                    _vibratorCount == 1 ? "VibrateCmd requires 1 vector for this device." :
+                        $"VibrateCmd requires between 1 and {_vibratorCount} vectors for this device.",
+                    Error.ErrorClass.ERROR_DEVICE,
+                    cmdMsg.Id);
+            }
+
+            foreach (var v in cmdMsg.Speeds)
+            {
+                if (v.Index >= _vibratorCount)
+                {
+                    return new Error(
+                        $"Index {v.Index} is out of bounds for VibrateCmd for this device.",
+                        Error.ErrorClass.ERROR_DEVICE,
+                        cmdMsg.Id);
+                }
+                byte speed = (byte)Math.Floor(v.Speed * 255);
+                _device.ControlOut(
+                    (0x02) << 5 | // Vendor Type
+                    (0x01) | // Interface Recipient
+                    (0x00), // Out Enpoint
+                    1,
+                    speed,
+                    0);
+
+            }
+
+            return new Ok(aMsg.Id);
+        }
+
+        private async Task<ButtplugMessage> HandleSingleMotorVibrateCmd(ButtplugDeviceMessage aMsg)
+        {
+            var cmdMsg = aMsg as SingleMotorVibrateCmd;
+            if (cmdMsg is null)
+            {
+                return BpLogger.LogErrorMsg(aMsg.Id, Error.ErrorClass.ERROR_DEVICE, "Wrong Handler");
+            }

```

Comment:
Is reshaper not turning the code directly after closed brackets into syntax errors? Of is it fxcop that does that? Either way, looks like linting isn't the same here as elsewhere.